pkg/transforms/conversion.go

```diff
@@ -73,3 +77,55 @@ func (f Conversion) TransformToJSON(edgexcontext *appcontext.Context, params ...
 	}
 	return false, errors.New("Unexpected type received")
 }
+
+// TransformToCloudEvent will transform a models.Event to a Cloud Event
+// It will return an error and stop the pipeline if a non-edgex event is received or if no data is received.
+func (f Conversion) TransformToCloudEvent(edgexcontext *appcontext.Context, params ...interface{}) (continuePipeline bool, stringType interface{}) {
+	if len(params) < 1 {
+		return false, errors.New("No Event Received")
+	}
+	edgexcontext.LoggingClient.Debug("Transforming to CloudEvent")
+	if result, ok := params[0].(models.Event); ok {
+		if len(result.Readings) == 0 {
+			return false, errors.New("No event readings to transform")
+		}
+
+		source := result.Device
+		event := cloudevents.NewEvent(cloudevents.VersionV1)
+		event.SetID(result.ID)
+		event.SetType(reflect.TypeOf(result).String())
+		event.SetSource(source)
+		// The whole event containing all of the readings is serialized into the cloud event.
+		if err := event.SetData(result); err != nil {
+			return false, fmt.Errorf("Error setting data field for cloud event, %s", err)
+		}
+		return true, event
+	}
+	return false, errors.New("Unexpected type received")
```

Comment:
For readability, check this first and return early. Then the bulk of the code doesn't have to be indented.